### PR TITLE
feat: LIVE-1939 change cosmos delegation flow and ledger validator

### PR DESCRIPTION
**Pull request by henrily-ledger**
### ❓ Context

- **Impacted projects**: [e.g. `live-desktop`, `@ledgerhq/live-common`]

LLM

- **Linked resource(s)**:

https://ledgerhq.atlassian.net/browse/LIVE-1939

_Write a clear and concise description of what this pull request is about and why it is needed._

### ✅ Checklist

- [ ] **Test coverage**: _Did you write any tests to cover the changes introduced by this pull request?_
- [x] **Atomic delivery**: _Is this pull request standalone? In order words, does it depend on nothing else?_
- [x] **No breaking changes**: _Does this pull request contain breaking changes of any kind? If so, please explain why._

### 📸 Demo

https://user-images.githubusercontent.com/33897120/169983616-20f2010f-697c-4608-b6ae-70249ba0d508.mp4

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-mobile/src/families/cosmos/shared/02-SelectAmount.js [hunk 1]
```diff
@@ -95,9 +95,17 @@ function DelegationAmount({ navigation, route }: Props) {
         ? validators.filter(v => !v.amount.eq(0))
         : validators;
 
-    const transaction = bridge.updateTransaction(tx, {
-      validators: filteredValidators,
-    });
+    const transaction = bridge.updateTransaction(
+      tx,
+      tx.mode === "delegate"
```
Comment (haammar-ledger) [inline apps/ledger-live-mobile/src/families/cosmos/shared/02-SelectAmount.js:100]: No need to handle other modes? `undelegate`, `redelegate`?

File: apps/ledger-live-mobile/src/families/cosmos/DelegationFlow/SelectValidator.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      });
+    },
+    [navigation, route.params],
+  );
+
+  const renderItem = useCallback(
+    ({ item }: { item: CosmosValidatorItem }) => (
+      <ValidatorRow account={account} validator={item} onPress={onItemPress} />
+    ),
+    [onItemPress],
```
Comment (haammar-ledger) [inline apps/ledger-live-mobile/src/families/cosmos/DelegationFlow/SelectValidator.tsx:51]: No need to include `account` here?

File: apps/ledger-live-mobile/src/families/cosmos/shared/02-SelectAmount.js (hunk 1 above)
Comment (henrily-ledger) [reply apps/ledger-live-mobile/src/families/cosmos/shared/02-SelectAmount.js:100]: undelegate and redelagate go into the else case, amounts on those goes with the filteredValidators

File: apps/ledger-live-mobile/src/families/cosmos/DelegationFlow/SelectValidator.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import { useSelector } from "react-redux";
+import { TrackScreen } from "../../../analytics";
+import { ScreenName } from "../../../const";
+import { accountScreenSelector } from "../../../reducers/accounts";
+import ValidatorHead from "../shared/ValidatorHead";
+import ValidatorRow from "../shared/ValidatorRow";
+
+type Props = {
+  account: AccountLike;
+  parentAccount?: Account;
```
[the remark this answers is not in the document]
Comment (henrily-ledger) [reply apps/ledger-live-mobile/src/families/cosmos/DelegationFlow/SelectValidator.tsx:18]: probably a bad copypaste, can remove them

**Pull request merged**